automation/release.sh

```diff
@@ -0,0 +1,90 @@
+
+    # install gh cli for uploading release artifacts, with prompt disabled to enforce non-interactive mode
+    gh_cli_dir=$(mktemp -d)
+    (
+        cd  "$gh_cli_dir/"
+        curl -sSL "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_linux_amd64.tar.gz" -o "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+        tar xvf "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    )
+    export PATH="$gh_cli_dir/gh_${GH_CLI_VERSION}_linux_amd64/bin:$PATH"
+    if ! command -V gh; then
+        echo "gh cli not installed successfully"
+        exit 1
+    fi
+    gh config set prompt disabled
+}
+
+function main() {
+
+    DOCKER_TAG="$(git tag --points-at HEAD | head -1)"
+    if [ -z "$DOCKER_TAG" ]; then
+        echo "commit $(git show -s --format=%h) doesn't have a tag, exiting..."
+        exit 0
+    fi
+    export DOCKER_TAG
+    export TRAVIS_TAG="$DOCKER_TAG"
+
+    ensure_gh_cli_installed
+
+    make
+    make build-verify # verify that we set version on the packages built by go (goveralls depends on go-build target)
+    make apidocs
+    make client-python
+    make manifests DOCKER_PREFIX="$DOCKER_PREFIX" DOCKER_TAG="$DOCKER_TAG" # skip getting old CSVs here (no QUAY_REPOSITORY), verification might fail because of stricter rules over time; falls back to latest if not on a tag
+    make olm-verify
+    make prom-rules-verify
+
+    make manifests
+    make build-functests
+
+    # add artifacts to created github release (overwrite existing)
+    gh auth login --with-token <"$GITHUB_TOKEN_PATH"
+    gh release upload --clobber "$DOCKER_TAG" _out/cmd/virtctl/virtctl-v* \
+        _out/manifests/release/demo-content.yaml \
+        _out/manifests/release/kubevirt-operator.yaml \
+        _out/manifests/release/kubevirt-cr.yaml \
+        _out/templates/manifests/release/kubevirt-operator.yaml.j2 \
+        _out/templates/manifests/release/kubevirt-cr.yaml.j2 \
+        _out/templates/manifests/release/demo-content.yaml.j2 \
+        _out/manifests/release/olm/kubevirt-operatorsource.yaml \
+        "_out/manifests/release/olm/bundle/kubevirtoperator.$DOCKER_TAG.clusterserviceversion.yaml" \
+        _out/tests/tests.test \
+        _out/manifests/release/conformance.yaml \
+        _out/templates/manifests/release/conformance.yaml.j2 \
+        _out/manifests/testing/*
+
+    # upload testing manifests to gcs (replaces periodic-kubevirt-update-release-x.y-testing-manifests periodics)
+    gsutil -m rm -r "gs://kubevirt-prow/devel/release/kubevirt/kubevirt/$DOCKER_TAG"
+    gsutil cp -r "_out/manifests/testing gs://kubevirt-prow/devel/release/kubevirt/kubevirt/$DOCKER_TAG/manifests/"
+
+    # generate latest stable version file
+    # will be available under http://storage.googleapis.com/kubevirt-prow/devel/release/kubevirt/kubevirt/stable.txt
+    export latest_stable_release_version="$(
+                curl --fail -s https://api.github.com/repos/kubevirt/kubevirt/releases |
```

Comment:
by not authenticating with GH to perform api calls like this. It's possible someday we could get rate limited if prow starts doing stuff like this for a lot of projects. 

logging in and using `gh release list` would probably be more reliable

also, maybe you know this. just wanted to point out curling the API returns limited results. the endpoint only returns 30 releases. given our current release cadence of a new stable release every month, i think we're fine.  using gh releases we have more control... but again. that's not even a concern for how kubevirt releases at the moment.

Reply:
That's a good point. This makes it also a bit easier to read, I'll change it to using `gh release list`.

Reply:
NB: `gh release list` by default returns the same amount of items (30). Unfortunately there's no pagination that we can use here, so at first I'll use ` -L 1000` which might be a bit off, but should keep us safe. Besides that the list appears to be sorted descending by release, although I'm not sure of that.